resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/k8s/features/VolcanoFeatureStep.scala

```diff
@@ -46,7 +46,7 @@ private[spark] class VolcanoFeatureStep extends KubernetesDriverCustomFeatureCon
         "for executor.")
       return Seq.empty
     }
-    val client = new DefaultVolcanoClient
+    lazy val client = new DefaultVolcanoClient
     val template = kubernetesConf.getOption(POD_GROUP_TEMPLATE_FILE_KEY)
     val pg = template.map(client.podGroups.load(_).item).getOrElse(new PodGroup())
```

Comment:
When `template` configuration is not given, Apache Spark will not create `DefaultVolcanoClient`

Reply:
Actually I think we should have better create the `DefaultVolcanoClient` within `map` but it's really a nit.

Reply:
Thank you. Yes, that line is the only place indeed. Let me merge this and take another look at the rest of test cases of this suite.